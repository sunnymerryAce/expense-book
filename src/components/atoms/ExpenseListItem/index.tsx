import React from "react";
import Typography from "@material-ui/core/Typography";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import Divider from "@material-ui/core/Divider";

export type ExpenseItemData = {
  categoryName: string;
  title: string;
  amount: number;
};
export interface ExpenseListItemProps extends ExpenseItemData {
  handleClickItem: (props: unknown) => unknown;
}

const ExpenseListItem: React.FC<ExpenseListItemProps> = (props) => {
  return (
    <>
      <ListItem
        button
        onClick={props.handleClickItem}
        className="ExpenseListItem"
      >
        <ListItemText
          primary={props.categoryName}
          secondary={props.title}
          className="ExpenseListItem-ListItemText"
        />
        <Typography
          variant="subtitle1"
          className="ExpenseListItem-amount"
        >{`Â¥${props.amount}`}</Typography>
      </ListItem>
      <Divider />
    </>
  );
};

export default ExpenseListItem;
