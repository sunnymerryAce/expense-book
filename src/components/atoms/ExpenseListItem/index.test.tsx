import React from "react";
import ExpenseListItem from "./";
import { shallow } from "enzyme";

describe("<ExpenseListItem />", () => {
  let container: any = null;
  const props = {
    category: "Cafe",
    name: "スタバ",
    amount: 300,
    date: "20200505T123456",
    handleClickItem: jest.fn(),
  };

  beforeEach(() => {
    container = shallow(<ExpenseListItem {...props} />);
  });

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  test("ListItem should have proper props", () => {
    expect(container.find(".ExpenseListItem").props()).toMatchObject({
      onClick: props.handleClickItem,
    });
  });

  test("ListItemText should have proper props", () => {
    expect(
      container.find(".ExpenseListItem-ListItemText").props()
    ).toMatchObject({
      primary: props.category,
      secondary: props.name,
    });
  });

  test("should show proper amount", () => {
    expect(container.find(".ExpenseListItem-amount").text()).toBe(
      `¥${props.amount}`
    );
  });

  test("click event handler should be triggered", () => {
    container.find(".ExpenseListItem").simulate("click");
    expect(props.handleClickItem).toBeCalled();
  });
});
