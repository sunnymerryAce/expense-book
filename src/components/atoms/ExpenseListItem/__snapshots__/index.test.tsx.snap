// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseListItem /> should match the snapshot 1`] = `"<div class=\\"MuiButtonBase-root MuiListItem-root ExpenseListItem MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div class=\\"MuiListItemText-root ExpenseListItem-ListItemText MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Cafe</span><p class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock\\">スタバ</p></div><h6 class=\\"MuiTypography-root ExpenseListItem-amount MuiTypography-subtitle1\\">¥300</h6></div><hr class=\\"MuiDivider-root\\"/>"`;
